### PR TITLE
layers: Add support for QCOM_image_processing

### PR DESCRIPTION
**Pull request by jackohound**
Adds ValidationLayer support for the vendor extension and adds corresponding tests

### CODE REVIEW

**Review by ncesario-lunarg — COMMENTED** [review]
A couple minor nits, but this LGTM!

The big one is the `[` character at the beginning of some VUID strings, as that is [currently blocking CI](https://github.com/KhronosGroup/Vulkan-ValidationLayers/actions/runs/3160162721/jobs/5144289629#step:6:230). It looks like we do not have any hardware capable of exercising this test, but also looks like the VUIDs that have this problem are not being tested.

File: scripts/spirv_validation_generator.py [hunk 1]
```diff
@@ -144,6 +144,7 @@ def __init__(self,
             {'vulkan' : 'VkPhysicalDeviceShaderIntegerDotProductFeaturesKHR', 'layer' : 'shader_integer_dot_product_features'},
             {'vulkan' : 'VkPhysicalDeviceShaderSubgroupUniformControlFlowFeaturesKHR', 'layer' : 'shader_subgroup_uniform_control_flow_features'},
             {'vulkan' : 'VkPhysicalDeviceRayTracingMaintenance1FeaturesKHR', 'layer' : 'ray_tracing_maintenance1_features'},
+            {'vulkan' : 'VkPhysicalDeviceImageProcessingFeaturesQCOM', 'layer' : 'shader_image_processing_features'},
```
Comment (sjfricke) [inline scripts/spirv_validation_generator.py:147]: 1. this should match the way you have it in `state_tracker.h` (**image_processing**_props) for consistency
2. You will need to add this variable in the bottom of `struct DeviceFeatures {` in `device_state.h`
3. run `python3 ./scripts/generate_source.py ./external/Vulkan-Headers/registry/ ./external/SPIRV-Headers/include/spirv/unified1/` to see this appear in `layers/generated/spirv_validation_helper.cpp` or else it won't be checking for the feature bit
4. remove the following above in this file before running the script

```patch
 self.capabilityExcludeList = [
-            'TextureBlockMatchQCOM',
-            'TextureBoxFilterQCOM',
-            'TextureSampleWeightedQCOM',
        ]
 ```

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 2]
```diff
@@ -16541,3 +16541,212 @@ TEST_F(VkLayerTest, AttachmentFeedbackLoopLayoutFeature) {
...
+
+    ivswci_bad.numPhases = image_proc_properties.maxWeightFilterPhases + 1;
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkImageViewSampleWeightCreateInfoQCOM-numPhases-06962");
+    m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkImageViewCreateInfo-pNext-06954");
+    CreateImageViewTest(*this, &ivci_bad, "VUID-VkImageViewSampleWeightCreateInfoQCOM-numPhases-06963");
+    ivswci_bad.filterSize.width = ivswci.filterSize.width;
+
+    VkImageView weight_image_view = {};
+    VkResult res = vk::CreateImageView(m_device->device(), &ivci, nullptr, &weight_image_view);
+    ASSERT_TRUE(res == VK_SUCCESS);
```
[the remark this answers is not in the document]
Comment (jackohound) [reply tests/vklayertests_buffer_image_memory_sampler.cpp:16703]: Ran into compatibility trouble using the OneOffDescriptorSet() in combination with the above wrapper.  Will keep this in mind for future.

File: scripts/spirv_validation_generator.py (hunk 1 above)
Comment (jackohound) [reply scripts/spirv_validation_generator.py]: The above capability bits in the process of being contributed, but are not yet in the spirv-headers, so I kept the ExcludeList for now.   Will follow up with that change afterwards.

File: layers/buffer_validation.cpp [hunk 3]
```diff
@@ -6492,6 +6494,170 @@ bool CoreChecks::PreCallValidateCreateImageView(VkDevice device, const VkImageVi
...
+                    "but the component swizzles ( r = %s, g = %s, b = %s, a = %s)"
+                    " are not VK_COMPONENT_SWIZZLE_IDENTITY for all components.",
+                    string_VkComponentSwizzle(pCreateInfo->components.r), string_VkComponentSwizzle(pCreateInfo->components.g),
+                    string_VkComponentSwizzle(pCreateInfo->components.b), string_VkComponentSwizzle(pCreateInfo->components.a));
+            }
+            if (aspect_mask != VK_IMAGE_ASPECT_COLOR_BIT) {
+                skip |= LogError(pCreateInfo->image, "VUID-VkImageViewCreateInfo-pNext-06947",
+                                 "vkCreateImageView(): pNext chain includes VkImageViewSampleWeightCreateInfoQCOM "
+                                 "but the subresource range aspect mask (%s) is not VK_IMAGE_ASPECT_COLOR_BIT.",
+                                 string_VkImageAspectFlags(aspect_mask));
```
Comment (ncesario-lunarg) [inline layers/buffer_validation.cpp:6525]: I think you need to call `.c_str()` on this and similar usage below.

**Review by ncesario-lunarg — APPROVED** [review]
LGTM if @sjfricke's concerns are addressed. Just the compilation issue.

File: scripts/spirv_validation_generator.py (hunk 1 above)
Comment (sjfricke) [reply scripts/spirv_validation_generator.py]: :+1:

**Comment by jackohound** [comment]
All feedback addressed.  CI is passing.  Ready to merge.

**Comment by ncesario-lunarg** [comment]
Sorry I missed this @jackohound. I think the conflicts are pretty trivial if you can do a quick rebase and pull the test code into a separate commit (which I missed during review), I can merge this.

**Comment by spencer-lunarg** [comment]
@jackohound 

> CI Vulkan-ValidationLayers build # 9070 failed.

This was a hiccup, is good to go. Looked at the merge conflict, just and issue of another PR writing to the bottom of `vklayertests_buffer_image_memory_sampler.cpp` and should be easy to resolve

**Comment by spencer-lunarg** [comment]
closing in favor of https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5536

**Pull request closed without merging**